### PR TITLE
pam_limits: adds both check and diff modes

### LINKED ISSUE #827: pam_limits: add check mode
##### SUMMARY

**pam_limits** module has no check mode.

##### ISSUE TYPE
- Feature Idea

##### COMPONENT NAME

pam_limits

### LINKED ISSUE #828: pam_limits: add diff mode
##### SUMMARY

**pam_limits** module has no diff mode.

##### ISSUE TYPE
- Feature Idea

##### COMPONENT NAME

pam_limits

### PR DESCRIPTION
**Pull request by wookietreiber**
##### SUMMARY

pam_limits: adds both check and diff modes

Fixes #827, fixes #828.

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME
pam_limits

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

File: plugins/modules/system/pam_limits.py [hunk 1]
```diff
@@ -289,16 +290,25 @@ def main():
...
-    module.atomic_move(nf.name, f.name)
+    with open(limits_conf, 'r') as content:
+        content_current = content.read()
+
+    with open(nf.name, 'r') as content:
+        content_new = content.read()
+
+    if not module.check_mode:
+        # Copy tempfile to newfile
+        module.atomic_move(nf.name, f.name)
```
Comment (felixfontein) [inline plugins/modules/system/pam_limits.py:301]: It's probably better to use `limits_conf` here as well:
```suggestion
        module.atomic_move(nf.name, limits_conf)
```

**Review by felixfontein — COMMENTED** [review]
You need a [changelog fragment](https://docs.ansible.com/ansible/devel/community/development_process.html#changelogs-how-to).

File: plugins/modules/system/pam_limits.py (hunk 1 above)
Comment (wookietreiber) [reply plugins/modules/system/pam_limits.py]: Done.

**Comment by wookietreiber** [comment]
I added the changelog fragments.

File: changelogs/fragments/827-pam-limits-check-mode.yml [hunk 2]
```diff
@@ -0,0 +1,2 @@
+minor_changes:
```
Comment (felixfontein) [inline changelogs/fragments/827-pam-limits-check-mode.yml:1]: Please put both fragments in the same file (two items in the `minor_changes` list), and change the number in the filename to 830 (which is this PR's #).

File: changelogs/fragments/830-pam-limits.yml [hunk 3]
```diff
@@ -0,0 +1,3 @@
+minor_changes:
+  - pam_limits - adds check mode (https://github.com/ansible-collections/community.general/issues/827)
+  - pam_limits - adds diff mode (https://github.com/ansible-collections/community.general/issues/828)
```
Comment (Andersson007) [inline changelogs/fragments/830-pam-limits.yml:3]:
```suggestion
  - pam_limits - adds check mode (https://github.com/ansible-collections/community.general/issues/827).
  - pam_limits - adds diff mode (https://github.com/ansible-collections/community.general/issues/828).
```
according to the doc

Comment (wookietreiber) [reply changelogs/fragments/830-pam-limits.yml]: done

**Comment by Andersson007** [comment]
i'd be happy to merge this but we should have this module tested manually by anybody as there are no integration tests for it.

**Comment by wookietreiber** [comment]
> i'd be happy to merge this but we should have this module tested manually by anybody as there are no integration tests for it.

I would have liked to include tests for these features, but I don't have the slightest idea where and how to start because I'm completely unfamiliar with the testing framework. Are there adaptable tests that check for the existence and/or correctness of check and diff mode in other modules? Some pointers would be appreciated.

**Comment by Andersson007** [comment]
all currently existing tests live in `tests/` subdirectories, you could look through them. In case of integration tests, they are just playbooks with asserts. I'm not sure it's possible to implement them for the case - if not, manual testing would be also ok.

**Comment by Akasurde** [comment]
recheck

**Comment by Akasurde** [comment]
@felixfontein @Andersson007 I added tests for this change in https://github.com/ansible-collections/community.general/pull/1575. Let us merge this and then #1575.

**Review by felixfontein — APPROVED** [review]

**Comment by felixfontein** [comment]
Restarting CI.

**Comment by felixfontein** [comment]
@wookietreiber thanks for implementing this!
@Andersson007 thanks for reviewing!
@Akasurde thanks for reviewing and implementing tests!

**Comment by Andersson007** [comment]
@wookietreiber would you like to be added as a maintainer of the module to `.github/BOTMETA.yml`?
You'll receive notifications about related issues / PRs and your `shipit`/`LGTM`/`+1` will be counted by bot for automerge (nothing obligatory from your side of course). What do you think?

**Pull request merged**